src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,248 @@
+    prefixes = set(col.split("_")[0] for col in df.columns if "_" in col)
+    for prefix in prefixes:
+        relevant_cols = [col for col in df.columns if col.startswith(prefix)]
+        mean_values = df[relevant_cols].mean(axis=1)
+        df[prefix + "_mean"] = mean_values
+        std_values = df[relevant_cols].std(axis=1)
+        df[prefix + "_std"] = std_values
+        df = df.drop(relevant_cols, axis=1)
+    return df
+
+
+def plot_data(df: pd.DataFrame, variable: str) -> None:
+    """
+    Plots the annual mean and dispersion of a specified variable for each city in \
+    the DataFrame.
+    :param df: The DataFrame containing the data to plot.
+    :param variable: The variable to plot.
+    :return: None
+    """
+
+    # Create a mapping from short variable names to full names
+    short_to_full = {v.split("_")[0]: v for v in VARIABLES}
+
+    # Retrieve the full name of the variable
+    full_variable_name = short_to_full.get(variable, variable)
+
+    plt.figure(figsize=(12, 8))
+
+    # Create a distinct color for each city
+    colors = plt.cm.viridis(np.linspace(0, 1, len(COORDINATES)))
+
+    for idx, city in enumerate(COORDINATES.keys()):
+        city_data = df[df["city"] == city]
+        mean = city_data[f"{variable}_mean"]
+        std = city_data[f"{variable}_std"]
+        plt.plot(
+            city_data["year"],
+            mean,
+            label=city,
+            color=colors[idx],
+            marker="o",
+            linestyle="-",
+            linewidth=2,
+        )
+        plt.fill_between(
+            city_data["year"], mean - std, mean + std, color=colors[idx], alpha=0.3
+        )
+
+    plt.title(
+        f'Annual Mean and Dispersion of {full_variable_name.replace("_", " ").capitalize()}',
+        fontsize=14,
+    )
+    plt.xlabel("Year", fontsize=12)
+    plt.ylabel(full_variable_name.replace("_", " ").capitalize(), fontsize=12)
+    plt.xticks(fontsize=10)
+    plt.yticks(fontsize=10)
+    plt.grid(True)
+    plt.legend(loc="upper left", bbox_to_anchor=(1, 1), fontsize=10)
+    plt.tight_layout()
+    plt.show()
+
 
 def main():
-    raise NotImplementedError
+    list_data_cities = []
+    for city in COORDINATES.keys():
+        logging.info(f"Getting data for {city}")
+        list_data_variables = []
+        for variable in VARIABLES:
+            logging.info(f"Getting data for {variable} in {city}")
+            for model in MODELS:
+                logging.info(f"Getting data for {model} in {city}")
+                data = get_data_meteo_api(
+                    city, "1950-01-01", "1960-12-31", variable, model
+                )
+                logging.info(data.head())
+                list_data_variables.append(data)
+        combined_df = list_data_variables[0]
+        for i, df in enumerate(list_data_variables[1:], 1):
+            combined_df = pd.merge(
+                combined_df, df, on=["time", "city"], suffixes=("", f"_df{i}")
```

Comment:
This could be done more elegantly. The city is the same for all. You could set time as index for example and use concat. Or concat axis=0 and then pivot.